### DISCUSSION TITLE
About the SnippetPBF Case，added a Prism but Visualization Is Not Working Properly

### DISCUSSION THREAD

**Discussion by 123uiui** [NVIDIA-Omniverse/PhysX#337] [Q&A]
Here is the code I added to this case, and it threw the exception shown in the image below.
![1](https://github.com/user-attachments/assets/ec567819-47cf-4219-92c6-89857c4797ef)
![2](https://github.com/user-attachments/assets/ae9ee2de-2b9b-4c27-a163-03918ea8f3a5)

`
static float randomVertices7[20][3] = {
	{ 0.875709, 0.000000, 0.000000 },
	{ 0.734122, 0.000000, 0.533371 },
	{ 0.313272, 0.000000, 0.964153 },
	{ -0.328454, 0.000000, 1.010877 },
	{ -0.863303, 0.000000, 0.627226 },
	{ -1.047619, 0.000000, 0.000000 },
	{ -0.836285, 0.000000, -0.607597 },
	{ -0.367963, 0.000000, -1.132475 },
	{ 0.338421, 0.000000, -1.041554 },
	{ 0.683973, 0.000000, -0.496935 },
	{ 0.836693, 0.1, 0.000000 },
	{ 0.799592, 0.1, 0.580938 },
	{ 0.314927, 0.1, 0.969246 },
	{ -0.297332, 0.1, 0.915093 },
	{ -0.693421, 0.1, 0.503800 },
	{ -0.851336, 0.1, 0.000000 },
	{ -0.796614, 0.1, -0.578774 },
	{ -0.309294, 0.1, -0.951908 },
	{ 0.307749, 0.1, -0.947155 },
	{ 0.869869, 0.1, -0.631997 }
};
`
`
static void createPrism()
{
	const PxVec3 * vertices = reinterpret_cast<const physx::PxVec3*>(&randomVertices7[0][0]);
	int verticesCount =20;
	physx::PxConvexMeshDesc convexDesc;
	convexDesc.points.count = verticesCount;
	convexDesc.points.stride = sizeof(PxVec3);
	convexDesc.points.data = vertices;
	convexDesc.flags =  PxConvexFlag::eCOMPUTE_CONVEX | PxConvexFlag::eFAST_INERTIA_COMPUTATION;
	//convexDesc.vertexLimit = 32;
	PxTolerancesScale scale;
	PxCookingParams params(scale);
	params.buildGPUData=true;
	PxConvexMesh* convexMesh = PxCreateConvexMesh(params, convexDesc,gPhysics->getPhysicsInsertionCallback());

	PxConvexMeshGeometry convexMeshGeometry = convexMesh;
	//convexMeshGeometry.scale.scale = 1.f;

	physx::PxShape* shape = PxGetPhysics().createShape(convexMeshGeometry, *gMaterial);

	PxTransform transform(PxVec3(0.0f, 0.0f, 0.0f), PxQuat(PxIdentity));
	physx::PxRigidDynamic* rigidDynamic = PxGetPhysics().createRigidDynamic(transform);
	rigidDynamic->setMass(10);
	//rigidDynamic->setRigidBodyFlag(PxRigidBodyFlag::eKINEMATIC, true);
	rigidDynamic->attachShape(*shape);
	gScene->addActor(*rigidDynamic);
	shape->release();
}
`

**Comment by vreutskyy** [comment]
Hi @123uiui 
You may need to enable [Debug Visualization](https://nvidia-omniverse.github.io/PhysX/physx/5.4.2/docs/DebugVisualization.html#debug-visualization)
```
scene->setVisualizationParameter(PxVisualizationParameter::eSCALE, 1.0f);
```

**Reply by 123uiui** [reply]
It didn't work.

**Reply by vreutskyy** [reply]
Aa, you mean snippet rendering, not debug visualization. Sorry, I don't see what exception you get, but if it worked before your modifications, then perhaps you're doing something wrong.

**Reply by 123uiui** [reply]
I did not modify the original code; I only added a prism convex mesh. This rigid body can be visualized correctly when rendered alone, but an error occurs when it is rendered together with the fluid.

**Reply by vreutskyy** [reply]
And if you remove the prism, does the fluid render normally?
I could have a look what's happening there if you provide a snippet/sample to reproduce the crash.
